onnx/shape_inference/implementation.cc

```diff
@@ -458,18 +464,20 @@ void InferShapes(
     const std::unordered_map<std::string, int>& opset_imports,
     const bool check_type,
     const ISchemaRegistry* schema_registry,
-    const int error_mode) {
+    const int error_mode,
+    const int data_prop) {
```

Comment:
it would be good to create an options struct
struct ShapeInferenceOptions {
bool check_types;
enum error_mode; or bool strict_error_mode;
bool enable_data_propagation;
}

Reply:
Good idea. Added